### PR TITLE
JSEA-22 : Fetch loans when component is loaded

### PR DESCRIPTION
**Pull request by JialinWang1**

### CODE REVIEW

File: front-end/src/components/LoanList/LoanList.jsx [hunk 1]
```diff
@@ -5,43 +5,49 @@ import PropTypes from 'prop-types'
...
-      </Content>
-      <FloatingButton
-        icon={{ type: 'Entypo', name: 'plus' }}
-        text="New Loan"
-        handlePress={handlePress}
-        buttonStyle={{ width: 160 }}
-      />
+      {loading ? (
+        <></>
+      ) : (
```
Comment (adamgyulavari) [inline front-end/src/components/LoanList/LoanList.jsx]: @JialinWang1 
```javascript
!loading && (...)
```

File: front-end/src/lib/URL.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const URL = {
+  getLoans: 'http://10.22.18.66:3000/loans'
```
Comment (adamgyulavari) [inline front-end/src/lib/URL.js]: @JialinWang1 why is this in a `lib` folder?

Comment (adamgyulavari) [reply front-end/src/lib/URL.js]: also extract the `baseUrl` into a constant and use this as a default value, but override it with `BASE_URL` env var

File: front-end/src/reducers/loanReducer.js [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
-    return data.loans
+    return { loading: true }
+  }
+  if (action.type === ACTION_TYPE.INIT_LOANLIST_SUCCESS) {
+    return { loanList: action.loanList, loading: false }
+  }
+  return {}
+}
+
+function fetchLoanList() {
```
Comment (adamgyulavari) [inline front-end/src/reducers/loanReducer.js]: @JialinWang1 this is an async action, it should be in the reducer

File: front-end/src/reducers/loanReducer.js [hunk 4]
```diff
@@ -1,11 +1,30 @@
...
+  return async dispatch => {
+    dispatch({ type: ACTION_TYPE.INIT_LOANLIST })
+    // eslint-disable-next-line no-undef
```
Comment (adamgyulavari) [inline front-end/src/reducers/loanReducer.js]: @JialinWang1 why is it needed?

File: front-end/src/reducers/loanReducer.js [hunk 5]
```diff
@@ -1,11 +1,30 @@
...
+    const data = await fetch(URL.getLoans, {
+      headers: { Authentication: 'Bearer token', Accept: 'application/json' }
+    }).then(res => res.json())
+    dispatch({ type: ACTION_TYPE.INIT_LOANLIST_SUCCESS, loanList: data.loanList })
+    // setTimeout(() => {
+    //   dispatch({ type: ACTION_TYPE.INIT_LOANLIST_SUCCESS })
+    // }, 1000)
```
Comment (adamgyulavari) [inline front-end/src/reducers/loanReducer.js]: @JialinWang1 outcommented code

File: front-end/src/components/LoanList/LoanCard/LoanCard.jsx [hunk 6]
```diff
@@ -17,7 +17,7 @@ function LoadCard({ loan }) {
           </Text>
           <Text style={style.loanType}>{loan.type.name}</Text>
           <Text style={style.interest}>{`Current Interest rate: ${loan.type.interest}%`}</Text>
-          <Text style={style.loanPrice}>{`$ ${loan.remaining}`}</Text>
+          <Text style={style.loanPrice}>{`$ ${loan.remaining.remaining}`}</Text>
```
Comment (adamgyulavari) [inline front-end/src/components/LoanList/LoanCard/LoanCard.jsx]: @JialinWang1 ?

File: front-end/src/reducers/rootReducer.js [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
+const initState = { loanList: [], productList: [], loading: true }
 
 // eslint-disable-next-line no-unused-vars
 const rootReducer = (state = initState, action) => {
   return {
     ...state,
-    loanList: loanReducer(action),
-    productList: productReducer(action)
+    ...loanReducer(action),
+    ...productReducer(action)
```
Comment (adamgyulavari) [inline front-end/src/reducers/rootReducer.js]: @JialinWang1 use `combineReducers` instead

File: front-end/tests/components/__snapshots__/LoanCard.test.js.snap [hunk 8]
```diff
@@ -128,7 +128,7 @@ exports[`<LoanHome /> <LoanList > renders correctly 1`] = `
         }
         uppercase={false}
       >
-        $ 3400
+        $ undefined
```
Comment (adamgyulavari) [inline front-end/tests/components/__snapshots__/LoanCard.test.js.snap]: @JialinWang1 these `undefined`-s doesnt look nice

File: front-end/package.json [hunk 9]
```diff
@@ -16,6 +16,7 @@
     "@unimodules/core": "^4.0.0",
     "expo": "^35.0.0",
     "expo-font": "~7.0.0",
+    "fetch-mock": "^8.0.0",
```
Comment (adamgyulavari) [inline front-end/package.json]: @JialinWang1 this should be in devDependencies

File: front-end/src/lib/actions.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
+/* global fetch */
```
Comment (adamgyulavari) [inline front-end/src/lib/actions.js]: @JialinWang1 let's put this into the .eslintrc instead

File: front-end/src/reducers/loanReducer.js [hunk 11]
```diff
@@ -1,11 +1,14 @@
-import data from '../../helpers/mockData_FE'
-import ACTION_TYPE from '../lib/actionType'
+import { ACTION_TYPE } from '../lib/actions'
 
-function loanReducer(action) {
-  if (action.type === ACTION_TYPE.INIT_LOANLIST) {
-    return data.loans
+const initState = { loanList: [] }
```
Comment (adamgyulavari) [inline front-end/src/reducers/loanReducer.js]: @JialinWang1 if there's a separate reducer for the loading, then here we just need the list
```suggestion
const initState = []
```

File: front-end/src/reducers/rootReducer.js [hunk 12]
```diff
@@ -1,15 +1,8 @@
+import { combineReducers } from 'redux'
 import loanReducer from './loanReducer'
-import productReducer from './productReducer'
+import loadingReducer from './loadingReducer'
+// import productReducer from './productReducer' for next task
```
Comment (adamgyulavari) [inline front-end/src/reducers/rootReducer.js]: @JialinWang1 I'm pretty sure whoever is going to that, can do the import without this help :D

File: front-end/tests/components/LoanList.test.js [hunk 13]
```diff
@@ -31,17 +43,31 @@ describe('<LoanList />', () => {
...
       component = renderer.create(
         <Provider store={store}>
           <LoanList />
         </Provider>
       )
     })
-    expect(store.dispatch).toBeCalledWith(expectedPayload)
+
+    // eslint-disable-next-line no-undef
+    window.fetch = jest.fn().mockImplementation(() =>
```
Comment (adamgyulavari) [inline front-end/tests/components/LoanList.test.js]: @JialinWang1 then why the need for `fetch-mock` library

File: front-end/src/reducers/rootReducer.js (hunk 12 above)
Comment (JialinWang1) [reply front-end/src/reducers/rootReducer.js]: that one is me

File: front-end/tests/components/LoanList.test.js (hunk 13 above)
Comment (JialinWang1) [reply front-end/tests/components/LoanList.test.js]: forgot to remove it

**Pull request merged**